Sources/SwiftDocC/Infrastructure/Workspace/LocalFileSystemDataProvider.swift

```diff
@@ -48,6 +48,11 @@ public struct LocalFileSystemDataProvider: DocumentationWorkspaceDataProvider, F
 
     public func contentsOfURL(_ url: URL) throws -> Data {
         precondition(url.isFileURL, "Unexpected non-file url '\(url)'.")
-        return try Data(contentsOf: url)
+        let fileHandle = try FileHandle(forReadingFrom: url)
+        guard let data = try fileHandle.readToEnd() else {
+            throw FileSystemError.noDataReadFromFile(path: url.path)
+        }
```

Comment:
If we have to do this instead of ` Data(contentsOf:)` everywhere in the future then I'm concerned that either we or someone new to the repo would forget and break things. 

Is there anyway that we can shim `Data(contentsOf:)` on non-Darwin platforms to do this or any way that we could mark  `Data(contentsOf:)` as unavailable so that we don't accidentally break things in the future?

Reply:
I'm not aware of a way to do that. There's no way to prevent someone from adding `import FoundationNetworking` in the future and shadowing the locally declared shim. The right way to do this is for swift-corelibs-foundation to deprecate this initializer, but that's up to its maintainers and whether they want to do the same thing on Darwin so that these different platform implementations don't diverge even more.

Reply:
Another thing that comes to mind is just checking for `FoundationNetworking` imports in `./bin/test` and raising an error if that's found. That will mean if anyone adds `Data(contentsOf:)` that will fail to build on Linux and Windows unless they add `import FoundationNetworking`, which in turn will make `./bin/test`.

Is `./bin/test` what you run on CI?

Reply:
Applied the latter suggestion in f2b590878ca7a19305d965572f722b26a39c8a58.

Reply:
>Is ./bin/test what you run on CI?
Yes